packages/core/rsc/streamable.tsx

```diff
@@ -235,6 +307,56 @@ export function createStreamableValue<T = any, E = any>(initialValue?: T) {
 
       warnUnclosedStream();
     },
+    /**
+     * This method is used to append a delta string to the current value. It
+     * requires the current value of the streamable to be a string.
+     *
+     * @example
+     * ```jsx
+     * const streamable = createStreamableValue('hello');
+     * streamable.append(' world');
+     *
+     * // The value will be 'hello world'
+     * ```
+     */
+    append(value: T) {
+      assertStream('.append()');
+
+      if (
+        typeof currentValue !== 'string' &&
+        typeof currentValue !== 'undefined'
+      ) {
+        throw new Error(
```

Comment:
would it make sense to have special AI errors that we can link to docs in the future? might be something for later